docs/apis/common/index.md

```diff
@@ -1350,3 +1350,397 @@ You may specify program arguments before the job is executed. The plan visualiza
 the execution plan before executing the Flink job.
 
 {% top %}
+
+Metrics
+-------------------
+
+Flink exposes a metric system that allows gathering and exposing metrics to external systems.
+
+### Registering metrics
+
+You can access the metric system from any user function that extends [RichFunction]({{ site.baseurl }}/apis/common/index.html#rich-functions) by calling `getRuntimeContext().getMetricGroup()`.
+This method returns a `MetricGroup` object on which you can create and register new metrics.
+
+### Metric types
+
+Flink supports `Counters`, `Gauges` and `Histograms`.
+
+#### Counter
+
+A `Counter` is used to count something. The current value can be in- or decremented using `inc()/inc(long n)` or `dec()/dec(long n)`.
+You can create and register a `Counter` by calling `counter(String name)` on a MetricGroup.
+
+{% highlight java %}
+
+public class MyMapper extends RichMapFunction<String, Integer> {
+  private Counter counter;
+
+  @Override
+  public void open(Configuration config) {
+    // create and register a counter
+    this.counter = getRuntimeContext().getMetricGroup().counter("myCounter");
+    ...
+  }
+
+  @public Integer map(String value) throws Exception {
+    // increment counter
+    this.counter.inc();
+    ...
+  }
+}
+
+{% endhighlight %}
+
+Alternatively you can also use your own `Counter` implementation:
+
+{% highlight java %}
+
+public class MyMapper extends RichMapFunction<String, Integer> {
+  ...
+
+  @Override
+  public void open(Configuration config) {
+    // register a custom counter
+    this.counter = getRuntimeContext().getmetricGroup().counter("myCustomCounter", new CustomCounter());
+    ...
+  }
+  ...
+}
+
+{% endhighlight %}
+
+#### Gauge
+
+A `Gauge` provides a value of any type on demand. In order to use a `Gauge` you must first create a class that implements the `org.apache.flink.metrics.Gauge` interface.
+There is not restriction for the type of the returned value.
```

Comment:
no, reporters generally use toString() on the value returned. that should probably be documented though.